radixdlt/babylon-wallet-android
# [ABW-2257] Add staking/unstaking tx preview

## Description
- add tx staking/unstaking preview
- rewrite tx review header and add animation

Note that header animation is laggy on debug build for no reason - checked recompostion count and does not seem to be huge. I run the build in release mode and animation is snappy there.
Stake claims are yet to come, this is why detekt fails with unused methods/parameters

## How to test

1. Run staking/unstaking transactions from dashboard
2. Run gumball scenarios with sandbox to see verify other tx types are not broken

## Video
https://github.com/radixdlt/babylon-wallet-android/assets/118203440/fe308838-3b58-4ec3-87f1-0fcf8c68f3cd

## Note 
I've added unstake rendering that is compatible with dashbaord created requests, but there should be more generic handling that can also handle multiple stake requests from same validator in one tx. 

## PR submission checklist
- [x] I have tested staking transaction rendering
- [x] I have run gumball tests to verify if other tx previews are not broken
